app/domain/processor.js

```diff
@@ -1,33 +1,51 @@
 const { ContainerClient } = require('@azure/storage-blob') // eslint-disable-line no-unused-vars
 const crypto = require('crypto')
-const { uploadChunkToBlob, withTimeout } = require('../services/blob-client')
 const { chunkDocument } = require('../utils/chunker')
+const { uploadDocument, deleteDocuments } = require('../services/azure-search-service')
+const { SearchClient } = require('@azure/search-documents') // eslint-disable-line no-unused-vars
+const { generateEmbedding } = require('../services/openai-service')
 
 /**
- * @typedef {{link: string, lastModified: string}} Manifest
+ * @typedef {{link: string, lastModified: string, documentKeys: string[]}} Manifest
  */
 
 /**
- * Chunks and uploads grants to azure blob storage
- * @param {{title: string, content: string, updateDate: Date, url: string}[]} grants
+ * Chunks and uploads grants to AI Search & removes old grants from AI Search
+ * @param {import('../services/govuk-api').Grant[]} grants
  * @param {Manifest[]} manifestGrants
  * @param {string} schemeName
  * @param {ContainerClient} containerClient
+ * @param {SearchClient} searchClient
  * @returns {{number, processedGrants: Manifest[]}}
  */
-const processGrants = async (grants, manifestGrants, schemeName, containerClient) => {
-  let chunkCount = 0
+const process = async (grants, manifestGrants, schemeName, containerClient, searchClient) => {
+  const removedGrants = manifestGrants.filter((manifestGrant) => isGrantRemoved(manifestGrant, grants))
+
+  await processRemovedGrants(removedGrants, searchClient)
+
+  await processGrants(grants, manifestGrants, schemeName, containerClient, searchClient)
+}
+
+/**
+ * Chunks and uploads grants to AI Search
+ * @param {import('../services/govuk-api').Grant[]} grants
+ * @param {Manifest[]} manifestGrants
+ * @param {string} schemeName
+ * @param {ContainerClient} containerClient
+ * @param {SearchClient} searchClient
+ * @returns {{number, processedGrants: Manifest[]}}
+ */
+const processGrants = async (grants, manifestGrants, schemeName, containerClient, searchClient) => {
   const processedGrants = []
 
-  for (const grant of grants) {
-    const lastModifiedString = manifestGrants.find((manifest) => manifest.link === grant.url)?.lastModified
-    const manifestDate = new Date(lastModifiedString)
+  let chunkCount = 0
 
-    // if the grant's last updated date is newer than manifest, add in the grant
-    // if this link is not present in the manifest, also add in the grant
-    if (!lastModifiedString || grant.updateDate > manifestDate) {
+  for (const [index, grant] of grants.entries()) {
```

Comment:
Is it worth wrapping this part in a 
```
try {
  for (const [index, grant] of grants.entries()) { }
} catch (err) {
  console.error('Error occurred at processGrants. Grants processed: ', chunkCount)
}

Reply:
Have added a try/catch within the for loop instead